units/user-configdrive.service

```
@@ -21,6 +21,7 @@ ConditionKernelCommandLine=!coreos.oem.id=digitalocean
 
 [Service]
 Type=oneshot
+ExecCondition=/usr/bin/bash -c "if [ -f '/etc/.ignition-result.json' ];then cat '/etc/.ignition-result.json' | /usr/bin/jq -e '.userConfigProvided == false';else exit 0;fi"
```

```suggestion
ExecCondition=/usr/bin/bash -c "if [ -f '/etc/.ignition-result.json' ];then cat '/etc/.ignition-result.json' | /usr/bin/jq -e '.userConfigProvided == false' ;else exit 1;fi"
```
I guess like this? Otherwise we would need a `&& exit 1` in the first branch

---

If the file is not there, should that not be an indication that the system was not configured by ignition and we should run?

---

Updated my suggestion, what I mean is that we have to explicitly call `exit` in the branch

---

Can we also avoid the cat and just feed the file directly to jq